### PR TITLE
Add support for AWS Linux and Debian

### PR DESCRIPTION
**Pull request by davidski**
Support the AWS Linux and Debian platforms. Centos not yet supported (requires EPEL).

### CODE REVIEW

File: recipes/install_package.rb [hunk 1]
```diff
@@ -19,7 +19,15 @@
...
-package 'r-base' do
+case node['platform_family']
+when 'rhel'
+  pkg_name = 'R'
+when 'debian'
+  pkg_name = 'r-base'
+else
+  pkg_name = 'r_base'
+end
+package pkg_name do
```
Comment (stevendanna) [inline recipes/install_package.rb]: newline after the `end`

File: recipes/install_package.rb [hunk 2]
```diff
@@ -19,7 +19,15 @@
 # limitations under the License.
 #
 
-package 'r-base' do
+case node['platform_family']
```
Comment (stevendanna) [inline recipes/install_package.rb]: For things like this where the case statement is choosing between data, I typically go for something like this:

```ruby
pkg_name = case node['platform_family']
           when 'rhel'
             ...
           end
```

**Comment by stevendanna** [comment]
Left some minor style comments.  Would you mind rebasing away the "Correct merge conflict" commit?  I can always do that when I merge so no worries if you don't get to it.

Is there not a dev package on Amazon/RHEL systems that we can install?

**Comment by davidski** [comment]
Thanks! Will clean this up soon.

There's no dev package that I've found in the Amazon repos, but I haven't checked out EPEL yet.

**Comment by davidski** [comment]
Sorry this took so long. All cleaned up and squashed now. Please review and merge if it passes muster. Thanks again!

**Comment by davidski** [comment]
Hi, @stevendanna. Will you have time to review and possibly merge this PR soon? I'd like to start working on the testing functions and would rather not have to track multiple PRs. :) Thanks for the continued engagement!

**Comment by stevendanna** [comment]
@davidski Heya.  I just pushed a basic travis setup.  I'm running your branch against our basic test-kitchen setup right now with one minor rubocop fix appended.  Once it passes, I'll merge it.

**Comment by stevendanna** [comment]
@davidski It looks for the current kitchen tests to pass on centos, we'd have to install EPEL (IIRC this is where the CRAN README's on centos support will send you to).  Since the `yum` cookbook already provides a yum_repository resource, it should be pretty straightforward.

**Comment by davidski** [comment]
Oy. I can't believe I had left this out so long. Trying to close out the year with a few less To Dos. :smile:

I've done a bit of refactoring on this PR.
- Ubuntu 10 dropped and Ubuntu 16 added
- rubocop/foodcritic/cookstyle linting
- Added the EPEL repository via the supermarket yum-epel recipe
- Updated Travis build environment for some ruby dependencies

Hopefully you'll be able to review and merge a bit faster than I was in getting this back to you. 😁

**Comment by sfromm-plm** [comment]
For centos-7.2, I had to change `default['r']['install_dir']` in `attributes/install.rb` to `'/usr/lib64/R'`.

**Comment by stevendanna** [comment]
@davidski Thanks for pulling this together.  I'll take a look and try to get another release out this month.

**Comment by davidski** [comment]
Hey there, @stevendanna. Might you be able to review and possibly merge this soon? Was dealing with a StackOverflow question recently where I think this would have been helpful. Thanks a bunch!

**Comment by stevendanna** [comment]
@davidski Yup, sorry I forgot about this.  I put it on my todo list for the week.

**Comment by stevendanna** [comment]
I went ahead and merged this as is, I'll try to take a look at this cookbook tomorrow and get it back in shape.

**Comment by davidski** [comment]
Sorry for missing this thread, @stevendanna! The Travis builds look to be passing (https://travis-ci.org/stevendanna/cookbook-r/builds/207778940 is green). Is there anything I'm missing here that I can help out with?

**Pull request merged**